storage/storage_image.go

```diff
@@ -844,6 +908,27 @@ func (s *storageImageDestination) commitLayer(ctx context.Context, blob manifest
 		return nil
 	}
 
+	s.lock.Lock()
+	diffOutput, ok := s.diffOutputs[blob.Digest]
+	s.lock.Unlock()
+	if ok {
+		layer, err := s.imageRef.transport.store.CreateLayer(id, lastLayer, nil, "", false, nil)
+		if err != nil {
+			return err
+		}
+
+		// FIXME: what to do with the uncompressed digest?
```

Comment:
@giuseppe Why is this (and the related `s.blobDiffIDs[blobDigest] = blobDigest` assignment), using the compressed values as the supposedly-unique uncompressed ones, correct and safe?